### DISCUSSION TITLE
Prisma middleware doesn't accept arbitrary data

### DISCUSSION THREAD

**Discussion by adesnmi** [prisma/prisma#8812] [Q&A] [answered]
I tried a Prisma integration with Honeycomb using middleware while waiting for the native implementation. I noticed that it's seemingly not possible to pass arbitrary data down into a middleware `$use` callback. Is there a reason for this?

For my use case, it means I can pass down a parent span id and trace the database calls that denote a query or mutation.

(A query/mutation in this context is a function with the potential of multiple database calls within them. The separation is purely conceptual)

Here's the best I've got

```ts
// integrations/honeycomb.ts
export const traceDb = (db: PrismaClient, honeycombContext: HoneyCombCtx) => {
  db.$use(async (params, next) => {
    const { action } = params
    const { tracer, spanContext } = honeycombContext
    const childSpan = tracer.startSpan(action, { childOf: spanContext })

    const result = await next(params)

    childSpan.finish()

    return result
  })
}
```

```ts
// app/foo/queries/bar.ts

if (ctx.honeycomb) {
  traceDb(db, ctx.honeycomb)
}

const product = await db.product.findUnique(...)

...
```

This works fine, however I would love to instead do:

```ts
// db/index.ts

import { PrismaClient } from "@prisma/client"

const client = new PrismaClient()

client.$use(async (params, next, userArgs) => {
    const { action } = params
    const { tracer, spanContext } = userArgs
    const childSpan = tracer.startSpan(action, { childOf: spanContext })

    const result = await next(params)

    childSpan.finish()

    return result
  })

export * from "@prisma/client"
export default client
```

```ts
// app/foo/queries/bar.ts

// Honeycomb is the third param. If given, it'll be injected into the $use call
const product = await db.product.findUnique(..., honeycomb)


```

I would love to know why the client doesn't accept arbitrary data; there are probably some implications I'm not thinking of.

**Comment by ryands17** [comment, marked as answer]
@muyiwaolu 👋 

The above workaround that you specified is the only way as of now. It would be great if you could create a [feature request](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) so that we can look into this 🙂

**Reply by adesnmi** [reply]
Thanks, I thought this was the only way but just wanted to double check

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
